lib/HTTP/Tiny.pm

```diff
@@ -1638,25 +1638,34 @@ sub can_reuse {
 }
 
 # Try to find a CA bundle to validate the SSL cert,
-# prefer Mozilla::CA or fallback to a system file
-sub _find_CA_file {
+# prefer IO::Socket::SSL::default_ca(), Mozilla::CA or fallback to a system file
+sub _find_CA {
     my $self = shift();
 
     my $ca_file =
       defined( $self->{SSL_options}->{SSL_ca_file} )
       ? $self->{SSL_options}->{SSL_ca_file}
-      : $ENV{SSL_CERT_FILE};
+      : undef;
 
     if ( defined $ca_file ) {
         unless ( -r $ca_file ) {
             die qq/SSL_ca_file '$ca_file' not found or not readable\n/;
         }
-        return $ca_file;
+        return ( SSL_ca_file => $ca_file );
```

Comment:
this patch changes the private API for HTTP::Tiny, returning a hash instead of a string containing the CA filename. Is this really necessary? While I understand that when it comes to fiddling with private methods _"if you break it you get to keep both parts"_, I think we shouldn't change something unless it gives us an advantage. Does it?

Reply:
This is because `default_ca()` can return both `SSL_ca_file` and/or `SSL_ca_path`, pointing to a ca-bundle file and a directory with certificates. 

An alternative could be to return nothing similar to how LWP::Protocol::https does, but this seemed clearer to me.

Reply:
Coming in long after the fact…

I think it does make sense to change internals to this new `_find_CA` with its new return signature. And/but the cost to avoid breaking anyone naughty is so minimal that I think we should do so:
````perl
# not for internal use; backcompat shim only
sub _find_CA_file { my %res = shift->_find_CA( @_ ); $res{SSL_ca_file} }
````
(If I haven’t misunderstood the change.)

Reply:
I've added the shim as suggested to HTTP::Tiny::Handle